File: frontend/app/src/main/java/com/example/geoquiz_frontend/data/remote/dtos/history/GameHistoryResponse.java
package com.example.geoquiz_frontend.data.remote.dtos.history;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class GameHistoryResponse {
    @SerializedName("matches")
    private List<GameSessionDto> matches;

    @SerializedName("totalCount")
    private int totalCount;

    @SerializedName("totalPages")
    private int totalPages;

    @SerializedName("totalWins")
    private int totalWins;

    @SerializedName("serverTime")
    private long serverTime;

    public List<GameSessionDto> getMatches() { return matches; }
    public int getTotalCount() { return totalCount; }
    public int getTotalPages() { return totalPages; }
    public int getTotalWins() { return totalWins; }
    public long getServerTime() { return serverTime; }
}

File: backend/GeoQuiz_backend/GeoQuiz_backend/Application/DTOs/Game/GameHistoryResponse.cs
namespace GeoQuiz_backend.Application.DTOs.Game
{
    public class GameHistoryResponse
    {
        public List<GameSessionDto> Matches { get; set; } = new List<GameSessionDto> ();
        public int TotalCount { get; set; }
        public int TotalPages { get; set; }
        public int TotalWins { get; set; }
        public long ServerTime { get; set; }
    }
}
